### ISSUE TITLE
Segmentation fault 11

### ISSUE THREAD

**Issue by theokirkland** [gpertea/stringtie#389]
I am trying to assemble four relatively small bam files and getting a segmentation fault: 11 error.  I can assemble two of them but the other two give this error.  The last few lines of the -v output before the fault is

[02/07 15:54:55]>bundle GG704912:61240-61389 [6 alignments (4 distinct), 0 junctions, 0 guides] begins processing...
[02/07 15:54:55]^bundle GG704912:61240-61389 done (0 processed potential transcripts).
[02/07 15:55:10]>bundle GG704912:61822-4648378 [8800860 alignments (1931765 distinct), 9640 junctions, 0 guides] begins processing...
Segmentation fault: 11

I am using a Mac M1 64 GB with Ventura 13.1 - I have run stringtie before and not had this problem.

I see from others that they could find  the problem in the bam file and eliminate it but I don't know how to do that.

Thanks

Theo Kirkland
UC San Diego
tkirkland@ucsd.edu

**Comment by gpertea** [comment]
That bundle does seem quite large and junction rich, though Stringtie should be able to handle it in 64GB I think.. Anyway,  I'm now quite curious to see what's going on there.

Is it possible to extract that bundle and send it to me for debugging ? See instructions here: https://github.com/gpertea/stringtie/wiki/Extracting-bundle-data-for-debugging 

It would be useful to also check if you can reproduce the crash on your M1 by running stringtie just on that bundle separately after you extracted it.

**Comment by gpertea** [comment]
thank you, got the files and was able to reproduce the problem (it crashes the same even on Linux). Noticed that assembling the two bundles separately works fine, only assembling them both at once triggers a crash, I'm going to look into that. 

A "bundle" is a cluster of overlapping read alignments in a genomic locus. I would not recommend discarding bundles unless there is a good reason to ignore reads aligned there which can reflect potentially valuable transcript information. 

Instead I would rather suggest the assembly each BAM file separately and merge the results. What is the reason you are feeding two BAM files at the same time to StringTie ? Are those two separate sequencing samples? 

I also noticed you are using version 2.1.6 instead of the latest v2.2.x version. Is there a particular technical reason for using that older version? I am planning to apply the fix just to the current version.

**Comment by theokirkland** [comment]
To be sure that I understand you - you would run string tie merge on multiple bam files - not stringtie.  Doesn’t stringtie merge require gif files?  Do you run stringtie on a single bam file to get a gif file.  My ultimate goal is to get a gtf file that has all the alignments from a total of 6 independent sequencing data sets.  What is the best way to do that?

Sorry for the naive questions.  I’m clearly at the edge of my competence and appreciate your help.

I thought that stringtie 2.1.6 was the most updated version.  I’ll update it.

Thanks again,

Theo

**Comment by gpertea** [comment]
Yes, see the protocol outlined at https://ccb.jhu.edu/software/stringtie/index.shtml?t=manual#de 

Yes, `stringtie --merge` takes as input the .gtf files which are the output of the assembly of individual sample (`.bam`) files. Alternatively, one can also use gffcompare for a much more conservative merging of those gtfs.

**Comment by theokirkland** [comment]
I really appreciate your help and detailed step by step instructions, which are exceptional.  I have used your workflow previously to analyze NGS data, including the publications, but temporarily forgot the lessons that I learned.  Thanks for the reminder.

Thanks again,

Theo
